crates/solver/src/settlement_submission/flashbots_settlement.rs

```diff
@@ -107,7 +107,7 @@ impl<'a> FlashbotsSolutionSubmitter<'a> {
         // mined previously submitted transaction time to propagate to our node.
 
         if !transactions.is_empty() {
-            const MINED_TX_PROPAGATE_TIME: Duration = Duration::from_secs(60);
+            const MINED_TX_PROPAGATE_TIME: Duration = Duration::from_secs(20);
```

Comment:
What I still don't understand is why would we want to "wait for mining" if the method errored or the deadline was reached? Why can't we immediately return in these two select cases? I get that we want to wait if the nonce changes.

Reply:
I am not the original author of this piece of code, but here is one case for which this is needed:
1. We submit tx to flashbots, and we start counting 10s pause before new loop iteration.
2. In the meantime, block A gets mined (not containing our tx)
3. After some time block B is mined (containing our tx)
4. Our node receives block A.
5. Our 10s is up but our node received only block A because of the delay in block propagation. We simulate tx and it fails, we return back
6. If we don't wait another 20s to receive block B, we wont see mined tx. 

Does this make sense?

Reply:
Yes, that makes perfect sense (I guess it depends on the type of error, we could probably detect an `OrderFilled` error, but that's advanced). Would you mind documenting that information somewhere in the code?

Reply:
Example added as a comment.